packit/api.py

```diff
@@ -75,25 +76,27 @@ def keep_fwding_ci_results(self) -> None:
         for topic, msg in self.consumerino.iterate_dg_pr_flags():
             self.process_ci_result(msg)
 
-    def update(self, dist_git_branch, dist_git_path: str = None):
+    def update(self, dist_git_branch: str):
         """
         Update given package in Fedora
         """
-        robot = PackitDistGitRobot(self.config, dist_git_path=dist_git_path)
-        full_version = robot.upstream_specfile.get_full_version()
+        dg = DistGit(self.config)
+        up = Upstream(self.config)
```

Comment:
What's wrong with this?
```python
def watch_something(...):
  for message in get_messages():
    config = Config.from_fedmsg_...(...)
    with PackitAPI(...) as api:
      api.do_something(config)
```

I agree though that it could be good to wrap that code in a class since it's gonna be very similar across all watch methods.

Reply:
Yes, that's what I mean.

Sync is built on top of the `PackitAPI`, I'll create a new class for this.

Reply:
I don't mean it's wrong, but to agree on the solution.